client/src/ts/animate.ts

```diff
@@ -0,0 +1,33 @@
+// Animation function for transitions between start and end using a step size.
+// the duration specifies the duration of the animation in seconds (60 fps).
+// the callback is called with the updated value each step.
+export function Animate(start: number, end: number, duration: number,
+                        step: number,
+                        callback: (new_val: number) => void): void {
+  let currentIteration = 1;
+  const interations = 60 * duration;
+  if (start > end) {
+    step = Math.abs(step) * -1;
+    }
```

Comment:
Done. Me too? Clang-format did that. I wonder if it doesn't play well with nested functions.